agent/src/main/scala/za/co/absa/atum/agent/AtumContext.scala

```diff
@@ -39,8 +39,11 @@ class AtumContext private[agent](
     parentAgent.getOrCreateAtumSubContext(atumPartitions ++ subPartitions)(this)
   }
 
-  def createCheckpoint(checkpointName: String, dataToMeasure: DataFrame) = {
-    ??? //TODO #26
+  def createCheckpoint(checkpointName: String, df: DataFrame): Unit = {
+    measures.foreach { measure =>
+      val result = MeasureResult(measure, measure.function(df))
+      AtumAgent.publish(checkpointName, this, result)
```

Comment:
Hmm so this will basically send many messages to the server, but for batch processing I think that all measurements should be sent in a single message, i.e. compute the results and gather them, and then call `publish`. This would require to change the existing API, or add another definition of the `publish` method, that would have multiple `measureResult` objects on its input.

Streaming will allow partial measurements, i.e. also sending a single measurement could make sense, and the server/DB would connect it together.

What do you think @benedeki ? Also, once my https://github.com/AbsaOSS/atum-service/pull/72 will be merged, you can directly work with `CheckpointDTO` object so this will be changed. But I would still like to talk about the concept of sending a single measurement vs multiple.

Reply:
I agree, I didn't want to guess the DAO format before it's merged, so this is done using the current dispatcher interface. After DAO is merged, I would prefer to change all this in different PR.

Reply:
all right then 👍

Reply:
I also agree with @lsulak.

Reply:
> I agree, I didn't want to guess the DAO format before it's merged, so this is done using the current dispatcher interface. After DAO is merged, I would prefer to change all this in different PR.

What's the benefit of an extra PR. This one is still tiny. 😉